src/core/utils/helpers/general.ts

```diff
@@ -481,23 +482,54 @@ export const getReleaseYearSearchResult = (work: Work) => {
   return getManifestationPublicationYear(latest) || "";
 };
 
-// Creates a "by author, author and author"-string
-export const getContributors = (
+export const createEtAlAuthors = (
   creators: string[],
-  by: string,
-  and: string
+  byText: string,
+  etAlText: string
 ) => {
-  let returnContentString = "";
+  const firstTwo = creators.slice(0, 2);
+  return `${byText} ${firstTwo.join(", ")} ${etAlText}`;
+};
+export const createAndAuthors = (
+  creators: string[],
+  byText: string,
+  andText: string
+) => {
+  return `${byText} ${creators
+    .slice(0, -1)
+    .join(", ")} ${andText} ${creators.slice(-1)}`;
+};
+
+// Creates a "by author, author and author"-string, or by author, author et al.
+export const getContributors = (short: boolean, creators: string[]) => {
+  // Todo this is sortof a hack, but using t: UseTextFunction as argument
+  // makes the components re-render.
+  const {
+    text: { data: texts }
+  } = store.getState();
```

Comment:
I really think we should avoid using the state directly. Either I think you should create it as a hook with `useSelector` to fetch text from the state or pass in the t function as a parameter.

Reply:
I've just moved it from one place to another. I did this a long time ago, because for reasons I could not figure out, if i sent the t function as a parameter endless loops would occur.